lib/chromedriver.js

```diff
@@ -625,11 +625,19 @@ class Chromedriver extends events.EventEmitter {
     }
 
     if (this.adb) {
-      log.debug(`Cleaning any old adb forwarded port socket connections`);
+      const udidIndex = this.adb.executable.defaultArgs.findIndex((item) => item === '-s');
+      const udid = udidIndex > -1 ? this.adb.executable.defaultArgs[udidIndex + 1] : null;
+
+      if (udid) {
+        log.debug(`Cleaning this device's adb forwarded port socket connections: ${udid}`);
+      } else {
+        log.debug(`Cleaning any old adb forwarded port socket connections`);
+      }
+
       try {
         for (let conn of await this.adb.getForwardList()) {
           // chromedriver will ask ADB to forward a port like "deviceId tcp:port localabstract:webview_devtools_remote_port"
-          if (conn.indexOf('webview_devtools') !== -1) {
+          if (conn.indexOf('webview_devtools') !== -1 && conn.indexOf(udid) !== -1) {
```

Comment:
Skip here when `udid` is null? (else clause in L634)

Reply:
prefer `includes` to `indexOf`